sdk/identity/Azure.Identity/src/DefaultAzureCredential.cs

```diff
@@ -155,9 +155,14 @@ private static async ValueTask<(AccessToken, TokenCredential)> GetTokenFromSourc
 
                     return (token, sources[i]);
                 }
+                catch (CredentialUnavailableException e)
+                {
+                    exceptions.Add(e);
+                }
                 catch (AuthenticationFailedException e)
                 {
                     exceptions.Add(e);
+                    break;
                 }
```

Comment:
I would remove this catch all together. The previous behavior was that exceptions which halt the authentication flow were thrown directly and the weren't aggregated, we shouldn't change that behavior here.